src/backend/linux_raw/mm/types.rs

```diff
@@ -111,6 +111,30 @@ bitflags! {
     }
 }
 
+impl MapFlags {
+    /// Create `MAP_HUGETLB` with provided size of huge page.
+    ///
+    /// Under the hood it computes `MAP_HUGETLB | (huge_page_size_log2 << MAP_HUGE_SHIFT)`.
+    /// `huge_page_size_log2` denotes logarithm of huge page size to use and should be
+    /// between 16 and 63 (inclusively).
+    ///
+    /// ```
+    /// use rustix::mm::MapFlags;
+    ///
+    /// let f = MapFlags::hugetlb_with_size(30).unwrap();
+    /// assert_eq!(f, MapFlags::HUGETLB | MapFlags::HUGE_1GB);
+    /// ```
+    pub const fn hugetlb_with_size(huge_page_size_log2: u32) -> Option<Self> {
```

Comment:
Could you name this something that indicates a log2 encoding, since "size" often means "size in bytes"? Perhaps with a "p2" in the name, following [`.p2align](https://ftp.gnu.org/old-gnu/Manuals/gas-2.9.1/html_node/as_111.html), so something like `huge_p2`?

Reply:
How about `hugetlb_with_log2_size` or `hugetlb_with_size_log2`, to make it consistent with the argument name? I don't think `p2` would be meaningful for most users.

Reply:
`hugetlb_with_size_log2` sounds reasonable to me. Or `huge_with_size_log2` since the specific constants like `MAP_HUGE_1GB` don't have the "tlb" in the name.

Reply:
I think `hugetlb_with_size_log2` is a better name since it includes the `MAP_HUGETLB` bit, while `MAP_HUGE_1GB` does not.